### PR TITLE
Validator client cleanup

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #5268 , this PR cleans up the validator clients handling of statuses in order to prepare for changes to WaitForActivation on the beacon client server.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Please split gRPC server and client changes into two different PRs.

**Comment by nisdas** [comment]
+1 , please split out the PRs

**Comment by 0xKiwi** [comment]
Cleaned up, only changes left are reorganizations on the validator client end.

File: validator/client/validator.go [hunk 1]
```diff
@@ -112,6 +112,37 @@ func (v *validator) WaitForChainStart(ctx context.Context) error {
 	return nil
 }
 
+// WaitForSync checks whether the beacon node has sync to the latest head
```
Comment (terencechain) [inline validator/client/validator.go:115]:
```suggestion
// WaitForSync checks whether the beacon node has sync to the latest head.
```

File: validator/client/validator.go [hunk 2]
```diff
@@ -112,6 +112,37 @@ func (v *validator) WaitForChainStart(ctx context.Context) error {
...
+	if err != nil {
+		return errors.Wrap(err, "could not get sync status")
+	}
+	if !s.Syncing {
+		return nil
+	}
+
+	for {
+		select {
+		// Poll every half slot
```
Comment (terencechain) [inline validator/client/validator.go:130]:
```suggestion
		// Poll every half epoch.
```

**Pull request merged**